benchmarks/entropy_adiabat/plugins/entropy_model.cc

```diff
@@ -0,0 +1,371 @@
+/*
+  Copyright (C) 2016 - 2020 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include "entropy_model.h"
+#include <aspect/adiabatic_conditions/interface.h>
+#include <aspect/utilities.h>
+
+#include <deal.II/base/table.h>
+#include <fstream>
+#include <iostream>
+
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    namespace MaterialUtilities
+    {
+      namespace Lookup
+      {
+        PerplexEntropyReader::PerplexEntropyReader(const std::string &filename,
+                                                   const MPI_Comm &comm)
+        {
+          /* Initializing variables */
+          interpolation = true;
+          delta_press=numbers::signaling_nan<double>();
+          min_press=std::numeric_limits<double>::max();
+          max_press=-std::numeric_limits<double>::max();
+          delta_temp=numbers::signaling_nan<double>();
+          min_temp=std::numeric_limits<double>::max();
+          max_temp=-std::numeric_limits<double>::max();
+          n_temperature=0;
+          n_pressure=0;
+
+          std::string temp;
+          // Read data from disk and distribute among processes
+          std::istringstream in(Utilities::read_and_distribute_file_content(filename, comm));
+
+          std::getline(in, temp); // eat first line
```

Comment:
I would find it useful to have a short comment about the data format here, so that it's clear what the purpose of all of this is. Also, isn't what you call `_temp` the entropy now?